### ISSUE TITLE
Web Interface Team

### ISSUE THREAD

**Issue by spitalnyd** [GatorCogitate/cogitate_tool#45]
This issue is used for Team Web Interface to discuss the creation of making a web outline for the entire project.  In addition to discuss any changes, additions, or updates to the web interface as we go.

**Comment by ABaldeosinghASC** [comment]
We can use streamlit to gather the data and display it as a webpage

**Comment by ABaldeosinghASC** [comment]
What data specifically are we putting through Streamlit though if anyone wants to clarify

**Comment by spitalnyd** [comment]
I believe we are using the features we made and using streamlit to make an web interface outline

**Comment by ABaldeosinghASC** [comment]
So the data that we collect through  pydriller is what we're going to pass into streamlit, correct ?

**Comment by Cory-Wiard** [comment]
@spitalnyd @ABaldeosinghASC @wilsonj24 @lussierc @reid7danny @TylerPham2000 Lets spend some time continuing to explore stream lit today and tomorrow, so that when we meet on thursday we can start and finish the implementation of it.

**Comment by spitalnyd** [comment]
Reminder we have a meeting at 4:30 today. This meeting will discuss the implementation of the web interface. Can't wait to see you all later today! Just let me know if you can't make it!!!

**Comment by TylerPham2000** [comment]
I have another meeting at 4:30 but I will come to this meeting as soon as possible.

**Comment by Cory-Wiard** [comment]
Hopefully during lab tomorrow we get a good understanding of how far people are, so that we can complete our part. Is everyone good with meeting Wednesday again if that is the case? @spitalnyd @ABaldeosinghASC @TylerPham2000 @wilsonj24 @lussierc @reid7danny

**Comment by lussierc** [comment]
I am good with meeting on Wednesday again. We can see what visualizations are most preferred by the customer today in order to determine what we will output in Streamlit. [Here is a helpful link for Streamlit API info & the types of visualizations supported.](https://docs.streamlit.io/api.html)

**Comment by lussierc** [comment]
@Cory-Wiard @wilsonj24 It looks like we are still waiting on the data/features from feature teams. After we get these we can implement these with our graphs that are currently using sample data. With this, I think we eventually move each graph to its own function.

**Comment by Cory-Wiard** [comment]
@lussierc Sounds good! After we implement the functions for each graph we can start testing the code as well to get our coverage up

**Comment by wilsonj24** [comment]
@Cory-Wiard @lussierc Great!

**Comment by lussierc** [comment]
Main Tasks left to complete for Web Interface:
- Have working graphs (most are complete and working, but with sample data since we don't have features yet)
- Get graphs to work with data from features (waiting on feature teams to complete these features and merge them into master so we can do this)
- Simple test cases for the web interface
   - Basically we can just test if the pandas data frame being used for the graph populates properly with the feature data

**Comment by lussierc** [comment]
1+ meeting(s) in the next two days (once we have features to actually use) to complete this.

**Comment by gkapfham** [comment]
What is the status on this issue @lussierc?

**Comment by lussierc** [comment]
Since the PR for the Web Interface has been merged into ‘master’, I am closing this Issue @gkapfham.
